### PR TITLE
added traffic tendecy

### PR DESCRIPTION
**Pull request by VysotskiVadim**
This is a private API, so absence of documentation is expected. 
This feature was already introduced in the past but then reverted: https://github.com/mapbox/mapbox-java/pull/1451

### CODE REVIEW

**Review by LukasPaczos — COMMENTED** [review]
@VysotskiVadim could you add source code documentation as well (not javadoc)?

File: samples/build.gradle [hunk 1]
```diff
@@ -30,6 +30,6 @@ dependencies {
 
 buildConfig {
     packageName = 'com.mapbox.sample'
-    buildConfigField 'String', 'MAPBOX_ACCESS_TOKEN', System.getenv("MAPBOX_ACCESS_TOKEN")
+    buildConfigField 'String', 'MAPBOX_ACCESS_TOKEN', project.hasProperty('MAPBOX_ACCESS_TOKEN') ? project.property('MAPBOX_ACCESS_TOKEN') : System.getenv('MAPBOX_ACCESS_TOKEN')
```
Comment (dzinad) [inline samples/build.gradle:33]: Should it be here?

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/DirectionsCriteria.java [hunk 2]
```diff
@@ -417,4 +440,17 @@ private DirectionsCriteria() {
...
+  @Retention(RetentionPolicy.CLASS)
+  @IntDef({
+    TRAFFIC_TENDENCY_UNKNOWN,
+    TRAFFIC_TENDENCY_CONSTANT_CONGESTION,
+    TRAFFIC_TENDENCY_INCREASING_CONGESTION,
+    TRAFFIC_TENDENCY_DECREASING_CONGESTION,
+    TRAFFIC_TENDENCY_RAPIDLY_INCREASING_CONGESTION,
+    TRAFFIC_TENDENCY_RAPIDLY_DECREASING_CONGESTION
+  })
+  public @interface TrafficTendencyCriteria {
```
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/DirectionsCriteria.java:454]: Is the annotation used anywhere?

Comment (VysotskiVadim) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/DirectionsCriteria.java:482]: only in comments

File: samples/build.gradle (hunk 1 above)
Comment (VysotskiVadim) [reply samples/build.gradle:33]: yes

**Comment by VysotskiVadim** [comment]
> @VysotskiVadim could you add source code documentation as well (not javadoc)?

Added. I only had doubts if I should mention new annotation in route options docs. I decided not to mention and not to add comments there

**Review by dzinad — APPROVED** [review]

**Pull request merged**